### PR TITLE
Added environment variables for controlling container's tag

### LINKED ISSUE #3: Config yaml documents all the latest version for release
so that docker-compose can use to update the latest

### PR DESCRIPTION
**Pull request by ywkim312**

### CODE REVIEW

File: docker/smile/docker-command-smile-traefik.sh [hunk 1]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export SENTIMENT_ANALYSIS_TAG=0.1.6
+export SCREEN_NAME_PROMPT_TAG=0.1.2
+export TOPIC_MODELING_TAG=0.1.4
+export NAME_ENTITY_RECOGNITION_TAG=0.1.3
+export AUTOPHRASE_TAG=0.1.6
+export IMAGE_CRAWLER_TAG=0.1.3
+export COLLECT_REDDIT_COMMENT_TAG=0.1.2
+
+# set component's tag for clowder
+export CLOWDER_UPLOAD_FILE_TAG=0.1.2
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: latest version is 0.1.3 now

File: docker/smile/docker-command-smile-traefik.sh [hunk 2]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export CLOWDER_LIST_TAG=0.1.0
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: 0.1.1

File: docker/smile/docker-command-smile-traefik.sh [hunk 3]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export CLOWDER_CREATE_SPACE_TAG=0.1.0
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: 0.1.1

File: docker/smile/docker-command-smile-traefik.sh [hunk 4]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export CLOWDER_CREATE_DATASET_TAG=0.1.0
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: 0.1.1

File: docker/smile/docker-command-smile-traefik.sh [hunk 5]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export CLASSIFICATION_SPLIT_TAG=0.1.2
+export CLASSIFICATION_TRAIN_TAG=0.1.2
+export HISTOGRAM_TAG=0.1.1
+export NETWORK_ANALYSIS_TAG=0.1.4
+export PREPROCESSING_TAG=0.1.4
+export SENTIMENT_ANALYSIS_TAG=0.1.6
+export SCREEN_NAME_PROMPT_TAG=0.1.2
+export TOPIC_MODELING_TAG=0.1.4
+export NAME_ENTITY_RECOGNITION_TAG=0.1.3
+export AUTOPHRASE_TAG=0.1.6
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: 0.1.7

File: docker/smile/docker-command-smile-traefik.sh [hunk 6]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
 #set server info
 export SERVER=smm.server.com
 
+# set each component's tag
+export SMILE_SERVER_TAG=0.3.2
+export SMILE_GRAPHQL_TAG=0.3.0
+export CLASSIFICATION_PREDICT_TAG=0.1.2
+export CLASSIFICATION_SPLIT_TAG=0.1.2
+export CLASSIFICATION_TRAIN_TAG=0.1.2
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: the 3 classification is at 0.1.3 now

File: docker/smile/docker-command-smile-traefik.sh [hunk 7]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export HISTOGRAM_TAG=0.1.1
+export NETWORK_ANALYSIS_TAG=0.1.4
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: 0.1.5

File: docker/smile/docker-command-smile-traefik.sh [hunk 8]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export PREPROCESSING_TAG=0.1.4
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: 0.1.5

File: docker/smile/docker-command-smile-traefik.sh [hunk 9]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export SENTIMENT_ANALYSIS_TAG=0.1.6
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: 0.1.7

File: docker/smile/docker-command-smile-traefik.sh [hunk 10]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export SCREEN_NAME_PROMPT_TAG=0.1.2
+export TOPIC_MODELING_TAG=0.1.4
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: 0.1.5

File: docker/smile/docker-command-smile-traefik.sh [hunk 11]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
...
+export NAME_ENTITY_RECOGNITION_TAG=0.1.3
```
Comment (longshuicy) [inline docker/smile/docker-command-smile-traefik.sh]: 0.1.4

File: docker/smile/docker-command-smile.sh [hunk 12]
```diff
@@ -13,6 +13,30 @@ export HOME=/your/home/directory
 #set server info
 export SERVER=smm.server.com
 
+# set each component's tag
```
Comment (longshuicy) [inline docker/smile/docker-command-smile.sh]: Same as the docker-command-smile-traefik.sh version might change.
These are the latest release
```
SMILE 0.3.2
GraphQL 0.3.0
clowder_create_collection 0.1.1
clowder_create_dataset 0.1.1
clowder_create_space 0.1.1
clowder_list 0.1.1
clowder_upload_file 0.1.3
autophrase 0.1.7
classification 0.1.3
image_crawler 0.1.4
name_entity_recognition 0.1.4
network_analysis 0.1.5
preprocessing 0.1.5
sentiment_analysis 0.1.7
topic_modeling 0.1.5
```

**Review by longshuicy — CHANGES_REQUESTED** [review]
Code looks good; I listed the latest version of each packages.
Could you test it on smm-tester if you have a chance to see if we've missed anything. Thanks!

File: docker/smile/docker-command-smile.sh (hunk 12 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 11 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 10 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 9 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 8 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 6 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 5 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 4 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 3 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 2 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/docker-command-smile-traefik.sh (hunk 1 above)
Comment (ywkim312) [reply docker/smile/docker-command-smile-traefik.sh]: updated

File: docker/smile/config.yaml [hunk 13]
```diff
@@ -0,0 +1,23 @@
+# set each component's tag
```
Comment (longshuicy) [inline docker/smile/config.yaml]: I dont think this is a valid yaml file, isn't it more like shell script? and you were calling it by "source xxx". Is it ok you rename it to be config.sh or something more suitable? Thanks!

Comment (ywkim312) [reply docker/smile/config.yaml]: true. let's change it to txt file. Since it is not a valid yaml file.

Comment (ywkim312) [reply docker/smile/config.yaml]: updated

**Comment by ywkim312** [comment]
> Code looks good; I listed the latest version of each packages. Could you test it on smm-tester if you have a chance to see if we've missed anything. Thanks!

tested in smm-tester. Basic test worked okay

**Review by longshuicy — APPROVED** [review]

**Review by longshuicy — APPROVED** [review]

**Pull request merged**